File: backend/src/EHub.Domain/Entities/ProjectProposal.cs
using System;
using System.Collections.Generic;
using EHub.Domain.Common;
using EHub.Domain.Enums;

namespace EHub.Domain.Entities;

public class ProjectProposal : AuditableEntity
{
    public Guid ProjectId { get; set; }
    public virtual Project Project { get; set; } = null!;

    public Guid TeamId { get; set; }
    public virtual Team Team { get; set; } = null!;

    public Guid ClassId { get; set; }
    public virtual Class Class { get; set; } = null!;

    public string Title { get; set; } = string.Empty;
    public string? StartupName { get; set; }
    public string? Tagline { get; set; }

    public string? Problem { get; set; }
    public string? Solution { get; set; }
    public string? TargetCustomers { get; set; }
    public string? ValueProposition { get; set; }
    public string? MarketSize { get; set; }
    public string? Competitors { get; set; }
    public string? BusinessModel { get; set; }
    public string? RevenueModel { get; set; }
    public string? MarketingStrategy { get; set; }
    public string? Technology { get; set; }
    public string? FinancialPlan { get; set; }
    public string? Roadmap { get; set; }
    public string? TeamIntroduction { get; set; }

    public ProjectProposalStatus Status { get; set; } = ProjectProposalStatus.Draft;

    public DateTime? SubmittedAt { get; set; }
    public DateTime? ApprovedAt { get; set; }
    public DateTime? RejectedAt { get; set; }

    public Guid? CreatedById { get; set; }
    public virtual User? Creator { get; set; }

    public Guid? UpdatedById { get; set; }
    public virtual User? Updater { get; set; }

    // Navigation properties
    public virtual ICollection<ProjectProposalVersion> Versions { get; set; } = new List<ProjectProposalVersion>();
    public virtual ICollection<ProjectComment> Comments { get; set; } = new List<ProjectComment>();
    public virtual ICollection<PitchDeck> PitchDecks { get; set; } = new List<PitchDeck>();
}

File: frontend/src/types/projectProposal.ts
export type ProjectProposalStatus =
  | 'Draft'
  | 'Submitted'
  | 'NeedsRevision'
  | 'Approved'
  | 'Rejected'
  | 'Archived';

export type ProjectProposalVersionPurpose = 'DraftSave' | 'Submission';

export type ProjectProposalAnalysisStatus = 'Pending' | 'Processing' | 'Completed' | 'Failed';
export type ProjectProposalOverlapRisk = 'InsufficientData' | 'Low' | 'Medium' | 'High';

export interface ProjectProposalContent {
  title: string;
  startupName: string;
  tagline: string;
  problem: string;
  solution: string;
  targetCustomers: string;
  valueProposition: string;
  marketSize: string;
  competitors: string;
  businessModel: string;
  revenueModel: string;
  marketingStrategy: string;
  technology: string;
  financialPlan: string;
  roadmap: string;
  teamIntroduction: string;
}

export interface ProjectProposalReview {
  id: string;
  proposalVersionId: string;
  fromStatus: ProjectProposalStatus;
  toStatus: ProjectProposalStatus;
  feedback: string;
  reviewedByUserId: string;
  occurredAtUtc: string;
}

export interface ProjectProposal extends ProjectProposalContent {
  id: string;
  projectId: string;
  teamId: string;
  classId: string;
  status: ProjectProposalStatus;
  currentSubmittedVersionId: string | null;
  currentAnalysisJobId: string | null;
  currentAnalysisStatus: ProjectProposalAnalysisStatus | null;
  submittedAtUtc: string | null;
  approvedAtUtc: string | null;
  rejectedAtUtc: string | null;
  rowVersion: string;
  reviews: ProjectProposalReview[];
}

export interface ProjectProposalDraft extends ProjectProposalContent {
  changeNote: string;
}

export interface CreateProjectProposalRequest extends ProjectProposalDraft {}

export interface UpdateProjectProposalRequest extends ProjectProposalDraft {
  rowVersion: string;
}

export interface SubmitProjectProposalRequest {
  rowVersion: string;
  changeNote: string;
}

export interface RestoreProjectProposalVersionRequest {
  rowVersion: string;
  changeNote: string;
}

export type ProjectProposalReviewDecision = 'Approved' | 'NeedsRevision' | 'Rejected';

export interface ReviewProjectProposalRequest {
  decision: ProjectProposalReviewDecision;
  feedback: string;
  rowVersion: string;
}

export interface ProjectProposalVersionSummary {
  id: string;
  versionNumber: number;
  purpose: ProjectProposalVersionPurpose;
  snapshotSchemaVersion: string;
  changeNote: string;
  changedByUserId: string;
  createdAtUtc: string;
}

export interface ProjectProposalVersion extends ProjectProposalVersionSummary {
  projectProposalId: string;
  snapshot: ProjectProposalContent;
}

export interface ProjectProposalAnalysisReport {
  summary: string;
  overlapRisk: ProjectProposalOverlapRisk;
  potentialDifferentiators: string[];
  limitations: string[];
  provider: string;
  model: string;
  promptVersion: string;
  outputSchemaVersion: string;
  embeddingProvider: string;
  embeddingModel: string;
  embeddingDimension: number;
  textSchemaVersion: string;
  retrievalVersion: string;
  matches: ProjectProposalAnalysisMatch[];
  generatedAtUtc: string;
}

export interface ProjectProposalAnalysisMatch {
  rank: number;
  proposalVersionId: string;
  projectProposalId: string;
  projectId: string;
  teamId: string;
  classId: string;
  classCode: string;
  semesterCode: string;
  title: string;
  startupName: string;
  semanticSimilarity: number;
  submittedAtUtc: string;
}

export interface ProjectProposalAnalysis {
  jobId: string;
  proposalVersionId: string;
  status: ProjectProposalAnalysisStatus;
  attemptCount: number;
  candidateScope: 'AllSystem';
  includeCrossSemester: boolean;
  languageMode: 'VietnameseAndEnglish';
  requestedAtUtc: string;
  processingStartedAtUtc: string | null;
  completedAtUtc: string | null;
  failedAtUtc: string | null;
  failureCode: string | null;
  canViewDetailedReport: boolean;
  report: ProjectProposalAnalysisReport | null;
}
